src/core/tsconfig.json

```diff
@@ -0,0 +1,42 @@
+{
+  "compilerOptions": {
+    "allowJs": true,
+    "checkJs": true,
+    "noEmit": true,
+    "downlevelIteration": true,
+    "target": "es5",
```

Comment:
Nit: is target necessary given we aren't using it for compilation?

Reply:
iiuc it sets some baselines around what syntax is supported or something. Every setting here was arrived at via "only add when nothing works and the internet says it's needed"

Reply:
Happy to look into this in a future PR, but I'm pretty sure its only need for transpilation: https://www.typescriptlang.org/tsconfig/#target